src/renderers/shared/reconciler/ReactMultiChild.js

```diff
@@ -276,8 +292,13 @@ var ReactMultiChild = {
      */
     _updateChildren: function(nextNestedChildrenElements, transaction, context) {
       var prevChildren = this._renderedChildren;
+      var removedChildren = {};
```

Comment:
> I believe that there is currently no allocation for leaves.

I'm not sure what you mean here? Components that have no children don't use ReactMultiChild at all right now. And reconciling an unchanged child does already allocate in flattenChildren. This does add one more though in the case of no changes.